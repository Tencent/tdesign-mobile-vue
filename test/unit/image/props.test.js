/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { mount } from '@vue/test-utils';
import Image from '../../../src/image';
import { h } from 'vue';

describe('Image', () => {
  describe(':props', () => {
    // 图片描述: 
    it('alt', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"","src":"https://tdesign.gtimg.com/site/upload1.png"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('error', () => {
      const fn = jest.fn();
      const wrapper = mount(Image, {
        propsData: {
          error: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('error', () => {
      const fn = jest.fn();
      const wrapper = mount(Image, {
        propsData: {
          error: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片填充模式: contain
    it('fit', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","fit":"contain"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片填充模式: cover
    it('fit', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","fit":"cover"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片填充模式: fill
    it('fit', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","fit":"fill"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片填充模式: none
    it('fit', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","fit":"none"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片填充模式: scale-down
    it('fit', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","fit":"scale-down"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否开启图片懒加载: false
    it('lazy', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","lazy":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
    it('loading', () => {
      const fn = jest.fn();
      const wrapper = mount(Image, {
        propsData: {
          loading: () => h('div', 'text')
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });

    it('loading', () => {
      const fn = jest.fn();
      const wrapper = mount(Image, {
        propsData: {
          loading: 'text'
        },
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 等同于原生的 object-position 属性，可选值为 top right bottom left 或 string，可以自定义任何px或者百分比: center
    it('position', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","position":"center"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片圆角类型: circle
    it('shape', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","shape":"circle"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片圆角类型: round
    it('shape', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","shape":"round"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 图片圆角类型: square
    it('shape', () => {
      const wrapper = mount(Image, {
        propsData: {"alt":"图片占位","src":"https://tdesign.gtimg.com/site/upload1.png","shape":"square"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
  });
});