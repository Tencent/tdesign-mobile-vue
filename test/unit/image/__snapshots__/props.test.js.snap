// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image :props alt 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt=""
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props error 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt=""
    class="t-image__img"
    src=""
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props error 2`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt=""
    class="t-image__img"
    src=""
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props fit 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: contain; object-position: center;"
  />
</div>
`;

exports[`Image :props fit 2`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: cover; object-position: center;"
  />
</div>
`;

exports[`Image :props fit 3`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props fit 4`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: none; object-position: center;"
  />
</div>
`;

exports[`Image :props fit 5`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: scale-down; object-position: center;"
  />
</div>
`;

exports[`Image :props lazy 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src=""
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props loading 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt=""
    class="t-image__img"
    src=""
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props loading 2`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt=""
    class="t-image__img"
    src=""
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props position 1`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props shape 1`] = `
<div
  class="t-image t-image--circle"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props shape 2`] = `
<div
  class="t-image t-image--round"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;

exports[`Image :props shape 3`] = `
<div
  class="t-image t-image--square"
>
  <div
    class="t-image__status"
  >
    <svg
      class="t-icon t-icon-ellipsis"
      fill="none"
      height="1em"
      viewBox="0 0 16 16"
      width="1em"
    >
      <path
        d="M3 9a1 1 0 110-2 1 1 0 010 2zM7 8a1 1 0 102 0 1 1 0 00-2 0zM12 8a1 1 0 102 0 1 1 0 00-2 0z"
        fill="currentColor"
        fill-opacity="0.9"
      />
    </svg>
  </div>
  <img
    alt="图片占位"
    class="t-image__img"
    src="https://tdesign.gtimg.com/site/upload1.png"
    style="object-fit: fill; object-position: center;"
  />
</div>
`;
